### PR TITLE
feat: add migrate-v2 command

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**

### CODE REVIEW

**Comment by shairez** [comment]
@dmitry-stepanenko the new package for city is `@qwik.dev/router` 😊

File: packages/qwik/src/cli/migrate-v2/run-migration.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    initialValue: true,
+  });
+
+  if (isCancel(proceed) || !proceed) {
+    bye();
+  }
+
+  try {
+    replacePackage('@builder.io/qwik-city', '@qwik.dev/city', versions['@qwik.dev/city']);
+    replacePackage('@builder.io/qwik', '@qwik.dev/qwik', versions['@qwik.dev/qwik']);
```
Comment (ianlet) [inline packages/qwik/src/cli/migrate-v2/run-migration.ts:28]: should we add `@builder.io/qwik-auth` -> `@auth/core` as well? considering it was removed in v2

Comment (PatrickJS) [reply packages/qwik/src/cli/migrate-v2/run-migration.ts:28]: yes please add

Comment (ianlet) [reply packages/qwik/src/cli/migrate-v2/run-migration.ts:28]: @shairez anything related to `qwik-labs` or other packages to add here?

**Comment by ianlet** [comment]
`QwikCityProvider` was renamed to `QwikRouterProvider`. Ideally, this renaming should be part of this migrate command as well.

File: packages/qwik/src/cli/migrate-v2/run-migration.ts (hunk 1 above)
Comment (shairez) [reply packages/qwik/src/cli/migrate-v2/run-migration.ts:28]: so... I think `qwik-auth` should be migrated to `@auth/qwik` but there are a bunch of extra migration steps for that, so I don't know if we want to get into that via this script.

Regarding `qwik-labs`, all of its features have been folded into either `core` or `router`.
But the only used one I know of is `insights`.

I've documented its changes here - 
https://github.com/QwikDev/qwik/pull/7018

again, it's not widely used, so I don't know if it worth the effort to make it automatic like the other more used ones.

**Comment by shairez** [comment]
Thanks @ianlet !

FYI qwik-react was renamed to `@qwik.dev/react`

**Comment by dmitry-stepanenko** [comment]
@shairez @PatrickJS @ianlet hey folks, WDYT if we change the target of this PR to `main`? I think that users should be able to install latest v1 and then run `npm run qwik migrate-v2`. It makes more sense than installing v2 manually

I also added logic to update the following names
```
QwikCityProvider => QwikRouterProvider
qwikCity => qwikRouter
QwikCityVitePluginOptions => QwikRouterVitePluginOptions
QwikCityPlugin => QwikRouterPlugin
createQwikCity => createQwikRouter
QwikCityNodeRequestOptions => QwikRouterNodeRequestOptions
```

Logic for the `@builder.io/qwik-auth` wasn't added, let's discuss it separately

**Comment by dmitry-stepanenko** [comment]
closing in favour of https://github.com/QwikDev/qwik/pull/6810, which targets v1 app

**Pull request closed without merging**